airflow/models/connection.py

```diff
@@ -278,16 +287,10 @@ def password(cls):
         """Password. The value is decrypted/encrypted when reading/setting the value."""
         return synonym('_password', descriptor=property(cls.get_password, cls.set_password))
 
-    def get_extra(self) -> Dict:
+    def get_extra(self) -> Optional[str]:
```

Comment:
```suggestion
    def get_extra(self) -> Optional[Dict]:
```

I think?

Reply:
No this should be str to begin with; this function is used to construct the declared_attr `extra`, which is in turn used to construct the property `extra_dejson`:

https://github.com/apache/airflow/blob/501a3c3fbefbcc0d6071a00eb101110fc4733e08/airflow/models/connection.py#L394=L408

It should be quite obvious `extra` can’t be a dict from how this property is implementation.

Reply:
But @dstandish recently implemented changes where extra_dejson is just left for compatibility but extra is supposed to be Dict - and whenever we see it is a string it will raise deprecation warnings.

https://github.com/apache/airflow/pull/21816

With the Airflow 3.0 vision that only Dict will be allowed.

But I agree `extra_dejson` suspiciously assume that extra is a string and I think it is used in a number of places - so  possibly it should be corrected even for 2.3.0 (and some `if "str" else assume dict and return it` should be implemented? @dstandish